vignettes/04-synthesis-data.Rmd

```diff
@@ -221,61 +188,45 @@ image_files <-
 image_files <- file.path("vignettes/", image_files)
 ```
 
-
 We will loop through these images, extract our plot data, and calculate the "greeness" of each extract.
 We are using the name of the file to extract the date for later.
 
 ```{r synth_get_greeness}
-library(raster)
-library(stringr)
-
 # Extract the date from the file name
 getDate <- function(file_name){
   date <- str_match_all(file_name, '[0-9]{4}-[0-9]{2}-[0-9]{2}')[[1]][,1]
   return(date)
 }
 
-
-# Get the clip coordinates into the correct order
-clip_coords <- paste(toString(bounding_box[4]),' ',toString(bounding_box[3]),
-                    ' ',toString(bounding_box[2]),' ',toString(bounding_box[1]))
-
-
-# Returns the greeness value of the clipped image
+# Returns the greeness value of the plot in the specified file
 getGreeness <- function(file_name, clip_coords){
-  out_file <- "extract.tif"
-
-  # Execute the GDAL command to extract the plot  
-  command = paste("gdal_translate -projwin ",clip_coords," ",file_name," ",out_file)
-  system(command)
-
-  # Load the red & green bands of the image and calculate the greeness value
-  red_image     <- raster(out_file, band = 1)
-  cellStats(red_image, stat = "mean")
+  band_image = raster(file_name, band = 1)
+  red_crop = crop(band_image, clip_coords)
 
-  green_image   <- raster(out_file, band = 2)
-  cellStats(green_image, stat = "mean")
+  band_image = raster(file_name, band = 2)
+  green_crop = crop(band_image, clip_coords)
 
-  add_rasters      <- green_image + red_image
+  add_rasters      <- green_crop + red_crop
   numerator        <- cellStats(add_rasters, stat = "sum")
 
-  subtract_rasters <- green_image - red_image
+  subtract_rasters <- green_crop - red_crop
   denominator      <- cellStats(subtract_rasters, stat = "sum")
 
   greeness         <- numerator / denominator
-
-  # Remove the temporary file
-  if (file.exists(out_file)) 
-    file.remove(out_file)
-
+  
+  # If we can't get a greeness value we return zero
+  if (is.nan(greeness)) {
+    greeness = 0;
+  }
+  
```

Comment:
This should probably return an NA, not a 0. If it's zero, it looks like actual data (i.e., the red and green rasters are the same).

Reply:
For a demo can't we avoid inputs that return nan?

Reply:
Two rasters currently return 0.

Reply:
I've removed the two offending files.